workflow/controller/agent.go

```diff
@@ -262,7 +262,9 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
 	if err != nil {
 		log.WithError(err).Info("Failed to create Agent pod")
 		if apierr.IsAlreadyExists(err) {
-			return created, nil
+			if existing, err := woc.controller.kubeclientset.CoreV1().Pods(woc.wf.ObjectMeta.Namespace).Get(ctx, pod.Name, metav1.GetOptions{}); err == nil {
```

Comment:
Why do we need another judgment here?  Is apierr.IsAlreadyExists enough?

Reply:
> created, err := woc.controller.kubeclientset.CoreV1().Pods(woc.wf.ObjectMeta.Namespace).Create(ctx, pod, metav1.CreateOptions{})

`created` will be nil even if the err returned is `AlreadyExists`.

Reply:
Were you hitting this case? Was it panicing when the nil result was referenced?